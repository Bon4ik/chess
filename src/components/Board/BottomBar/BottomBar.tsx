import React from "react";

import Button from "components/shared/Button";
import { BottomBarContainer } from "./BottomBar.styled";

interface IBottomBarComponent {
  addPawn: () => void;
  isAddPawnDisabled: boolean;
}

const BottomBarComponent: React.FC<IBottomBarComponent> = ({
  addPawn,
  isAddPawnDisabled,
}) => (
  <BottomBarContainer>
    <Button disabled={isAddPawnDisabled} onClick={addPawn}>
      Add pawn
    </Button>
  </BottomBarContainer>
);

export default BottomBarComponent;
