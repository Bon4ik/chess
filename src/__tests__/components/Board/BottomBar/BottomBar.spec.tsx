import React from "react";
import renderer from "react-test-renderer";
import BottomBar from "components/Board/BottomBar";

describe("BottomBar", () => {
  it("should renders correctly", () => {
    const givenProps = {
      addPawn: jest.fn(),
      isAddPawnDisabled: false,
    };
    const tree = renderer.create(<BottomBar {...givenProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
