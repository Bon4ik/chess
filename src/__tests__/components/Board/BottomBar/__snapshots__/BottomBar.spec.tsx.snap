// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomBar should renders correctly 1`] = `
<div
  className="sc-gtssRu cRKlEm"
>
  <button
    className="sc-bdnylx kfFHIt"
    disabled={false}
    onClick={[MockFunction]}
  >
    Add pawn
  </button>
</div>
`;
